packages/core-mobile/app/services/send/SendServiceBTC.ts

```diff
@@ -11,129 +8,76 @@ import {
-        }
-
-        return { inputs, outputs }
-      })
-  }
-
   async validateStateAndCalculateFees(
     params: ValidateStateAndCalculateFeesParams
   ): Promise<SendState> {
-    const { sendState, isMainnet, fromAddress, currency, sentryTrx } = params
+    const { sendState, isMainnet, fromAddress, sentryTrx, utxos } = params
+
     return SentryWrapper.createSpanFor(sentryTrx)
       .setContext('svc.send.btc.validate_and_calc_fees')
       .executeAsync(async () => {
-        const { amount, address: toAddress, defaultMaxFeePerGas } = sendState
-        const feeRate = Number(defaultMaxFeePerGas)
-        const amountInSatoshis = amount ?? 0n
-        const { utxos } = await this.getBalance({
-          isMainnet,
-          address: fromAddress,
-          currency: currency || '',
-          sentryTrx
-        })
-        const provider = getBitcoinProvider(!isMainnet)
+        const {
+          amount,
+          address: toAddress,
+          defaultMaxFeePerGas,
+          token
+        } = sendState
+
+        // Set canSubmit to false if token/address is not set
+        if (!token || !toAddress) return this.getErrorState(sendState, '')
 
         if (!defaultMaxFeePerGas || defaultMaxFeePerGas === 0n)
           return this.getErrorState(
-            {
-              ...sendState
-            },
+            sendState,
             SendErrorMessage.INVALID_NETWORK_FEE
           )
 
+        if (!utxos) {
+          return this.getErrorState(
+            sendState,
+            SendErrorMessage.BALANCE_NOT_FOUND
+          )
+        }
+
+        const feeRate = Number(defaultMaxFeePerGas)
+        const amountInSatoshis = amount ?? 0n
+
         // Estimate max send amount based on UTXOs and fee rate
-        // Since we are only using bech32 addresses we can use this.address to estimate
         const maxAmount = BigInt(
           Math.max(
-            getMaxTransferAmount(utxos, fromAddress, fromAddress, feeRate),
+            getMaxTransferAmount(utxos, toAddress, fromAddress, feeRate),
             0
           )
         )
 
-        if (!toAddress)
-          return this.getErrorState(
-            {
-              ...sendState,
-              maxAmount
-            },
-            SendErrorMessage.ADDRESS_REQUIRED
-          )
+        const newState: SendState = {
+          ...sendState,
+          canSubmit: true,
+          error: undefined,
+          maxAmount
+        }
 
         // Validate the destination address
         const isAddressValid = isBtcAddress(toAddress, isMainnet)
 
         if (!isAddressValid)
-          return this.getErrorState(
-            { ...sendState, canSubmit: false, maxAmount },
-            SendErrorMessage.INVALID_ADDRESS
-          )
-
-        const { fee, psbt } = createTransferTx(
-          toAddress,
-          fromAddress,
-          Number(amountInSatoshis),
-          feeRate,
-          utxos,
-          provider.getNetwork()
-        )
```

Comment:
there is no need to create an actual transaction object here anymore since the fee will be calculated in the bitcoin module already